### PR TITLE
treewide: All configureFlags are lists 

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

I have one more commit to enforce this, but not sure whether that's appropriate for 18.09.

I also sprinkled in `{enable,with}Feature{,As}` to make some things more concise.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @aneeshusa

### CODE REVIEW

File: pkgs/applications/editors/gobby/default.nix [hunk 1]
```diff
@@ -19,8 +19,7 @@ in stdenv.mkDerivation rec {
   buildInputs = [ gtkmm2 gsasl gtksourceview libxmlxx libinf intltool ]
     ++ stdenv.lib.optional gnomeSupport gnome_vfs;
   
-  configureFlags = ''
-  '';
+  configureFlags = [];
```
Comment (volth) [inline pkgs/applications/editors/gobby/default.nix]: may this line be removed?

Comment (Ericson2314) [reply pkgs/applications/editors/gobby/default.nix]: Yes I suppose so.

**Comment by matthewbauer** [comment]
I think scilab still has the issue: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/math/scilab/default.nix

**Comment by Ericson2314** [comment]
But I refactored that: https://github.com/ObsidianSystems/Nixpkgs/blob/configureFlags-cleanup/pkgs/applications/science/math/scilab/default.nix . OfBorg is confusing me.

**Comment by volth** [comment]
> I think scilab still has the issue: 

```scilab``` version is 10 years old and the derivation is marked broken 2 years old.
Maybe better to remove it

**Comment by Ericson2314** [comment]
@matthewbauer found the issue. Hopefully there's just a few of these. Better to remove it separately.

**Comment by Ericson2314** [comment]
LMK if this causes staging problems.

**Pull request merged**